DigitalLearningSolutions.Data.Tests/Services/CourseContentServiceTests.cs

```diff
@@ -36,5 +39,90 @@ public void Get_course_content_should_return_course()
+            const int candidateId = 9;
+            const int customisationId = 259;
+
+            // When
+            var result = courseContentService.GetProgressId(candidateId, customisationId);
+
+            // Then
+            const int expectedProgressId = 10;
+            result.Should().Be(expectedProgressId);
+        }
+
+        [Test]
+        public void Update_login_count_should_not_increment_login_if_no_new_session()
+        {
+            // Given
+            const int progressId = 10;
+            var expectedLoginCount = courseContentTestHelper.GetLoginCount(progressId);
+
+            // When
+            courseContentService.UpdateLoginCountAndDuration(progressId);
+            var result = courseContentTestHelper.GetLoginCount(progressId);
+
+            // Then
+            result.Should().Be(expectedLoginCount);
+        }
+
+        [Test]
+        public void Update_duration_should_not_increment_duration_if_no_new_session()
+        {
+            // Given
+            const int progressId = 10;
+            var expectedDuration = courseContentTestHelper.GetDuration(progressId);
+
+            // When
+            courseContentService.UpdateLoginCountAndDuration(progressId);
+            var result = courseContentTestHelper.GetDuration(progressId);
+
+            // Then
+            result.Should().Be(expectedDuration);
+        }
+
+        [Test]
+        public void Update_login_count_should_increment_login_if_new_session()
+        {
+            // Given
+            const int candidateId = 9;
+            const int customisationId = 259;
+            const int progressId = 10;
+            const int duration = 5;
+            var loginTime = new DateTime(2010, 9, 23);
+            var initialLoginCount = courseContentTestHelper.GetLoginCount(progressId);
+
+            // When
+            courseContentTestHelper.InsertSession(candidateId, customisationId, loginTime, duration);
+            courseContentService.UpdateLoginCountAndDuration(progressId);
+            var result = courseContentTestHelper.GetLoginCount(progressId);
+
+            // Then
+            result.Should().Be(initialLoginCount + 1);
+        }
+
+        [Test]
+        public void Update_duration_should_increment_duration_if_new_session()
+        {
+            // Given
+            const int candidateId = 9;
+            const int customisationId = 259;
+            const int progressId = 10;
+            const int duration = 5;
+            var loginTime = new DateTime(2010, 9, 23);
+            var initialDuration = courseContentTestHelper.GetDuration(progressId);
+
+            // When
+            courseContentTestHelper.InsertSession(candidateId, customisationId, loginTime, duration);
+            courseContentService.UpdateLoginCountAndDuration(progressId);
+            var result = courseContentTestHelper.GetDuration(progressId);
+
+            // Then
+            result.Should().Be(initialDuration + duration);

```

Comment:
Actually there's probably one more case we need to test; that the duration and login count don't get updated if the login time is not between the first submitted time and the submitted time. So I'd suggest adding two tests for that (one for duration and one for login) 🙂